python/bucketbase/ibucket.py

```diff
@@ -311,6 +327,8 @@ def put_object(self, name: PurePosixPath | str, content: Union[str, bytes, bytea
         :raises io.UnsupportedOperation: If the object already exists
         """
         self._lock_object(name)
+        if self._base_bucket.exists(name):
+            raise io.UnsupportedOperation(f"Object {name} already exists in AppendOnlySynchronizedBucket")
```

Comment:
Here's a BUG, since the Lock will remain opened, as on the previous line the lock is opened, and not closed here.